### ISSUE TITLE
Is it possible to load knowledge from a kjar dependency?

### ISSUE THREAD

**Issue by simonedavico** [apache/incubator-kie-examples#106]
Hi, we would like to migrate our KIE Execution Server setup to Kogito. We currently package our rules (composed of .drl and .gdst files) into a KJAR. The kmodule of the KJAR defines a default kie base and a default stateless kie session:

```
<kmodule xmlns="http://www.drools.org/xsd/kmodule" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <kbase name="kiebase1" default="true" eventProcessingMode="stream" equalsBehavior="identity">
    <ksession name="kiesession1" type="stateless" default="true" clockType="realtime"/>
  </kbase>
</kmodule>
```

When we try to retrieve the kie session,either as `SessionUnit` injected via `@Named` or via `KieRuntimeBuilder`, the kie session is not found.

If we copy the rules into the project resources and define the session in the resources `kmodule` everything starts working, at least partially. I say partially because DRL rules are fired, but GDST rules seem to be ignored.

In summary:

1. Is it possible to load rules from a KJAR in Kogito?
2. Are .gdst rules supported? I guess not, since they are not fired and your examples only include .xls for decisions table, but it would be great to have a confirmation.

Thank you for your support.

**Comment by evacchi** [comment]
Hi Simone

1. no, the deployment model for Kogito does not include (at least currently) support for kjars. My suggestion is to setup a fresh Kogito project and move your resources there
2. currently only DRL and XLS rules are supported

**Comment by simonedavico** [comment]
Thanks Edo, good to know :) given that we need to setup guided decision tables through Business Central, what would be the best way to extract drls from them to add them to our Kogito project without having to do a lot of manual work?

**Comment by simonedavico** [comment]
If anyone else stumbles into this, we worked around this issue by writing a maven plugin which loads knowledge files from a kjar at compile time, and also converts `.gdst` files to `.drl`.

You can check it out [here](https://github.com/welldsagl/kogito-kjar-maven-plugin).

**Comment by danipenaperez** [comment]
Hi all, 
@evacchi Do you know if will be available kjars support for Kogito soon?. I think is a good functionallity for packaging versioning and monitoring with KScanner.
